src/Microsoft.Health.Fhir.Core/Features/Search/SearchValues/UriSearchValue.cs

```diff
@@ -143,5 +143,10 @@ public override string ToString()
 
             return asString.ToString();
         }
+
+        public override int GetHashCode()
+        {
+            return Uri != null ? Uri.ToString().GetHashCode(StringComparison.Ordinal) : 0;
```

Comment:
@LTA-Thinking If I understand correctly, I should also use OriginialString here instead ToString()?

Reply:
Yes, I think so.

Reply:
Ah never mind, this Uri is already of type string so I don't need ToString or OriginalString :) Thanks!